cEP-0020.md

```diff
@@ -0,0 +1,529 @@
+  - Name: Beginner
+  - Points_required = 20
+  - Privileges:
+    - Get assigned to a difficulty low issue
+- Level-IV:
+  - Name: Intermediate
+  - Points_required = 50
+  - Privileges:
+    - Mark your own PR as wip
+- Level-V:
+  - Name: Cool
+  - Points_required = 100
+  - Privileges:
+    - Mark a PR by another newcomer as wip
+    - Get assigned to difficulty medium issue
+- Level-VII:
+  - Name: Expert
+  - Points_required = 200
+  - Privileges
+    - Assign newcomers to difficulty newcomer issue
+    - Assign newcomers to difficulty low issue
+- Level-VIII:
+  - Name: Legend
+  - Points_required = 350
+  - Privileges
+    - Invite newcomers to org
+    - Assign newcomers to difficulty medium issue
+    - Permission to `ack` and `unack` on the pull requests
+
+Implementation of privileges is beyond scope of the
+gamification system. The gamification system will merely
+expose a list of privileges that each person has, and other
+systems are expected to adapt to use these privileges.
+E.g.: [corobo](https://github.com/coala/corobo)
+
+While implementing privileges, a first manual check should
+be done for testing purpose, and then automated only if there
+is no serious risk or there is a good risk reaction mechanism
+in place.
+
+### Badges
+
+Another way of explicitly nudging a newcomer to action is to
+award badges for completing tasks. Badges enable newcomers to
+follow their performance and compare to other newcomers.
+Badges will be awarded after performing certain activities by
+the newcomers:
+
+- The bug finder:
+  - Details: The one who find bugs in the existing codebase
+  - Activities:
+    - Create at least one issue which labelled as bug
+- The bear hunter:
+  - Details: The one who create issues about new bears
+  - Activities:
+    - Create at least one bear proposal issue
+- The bear writer:
+  - Details: The one who write new bears
+  - Activities:
+    - Merged at least 2 bear proposal PRs
+- The helper:
+  - Details: The one who answers every newcomer questions on Gitter
+  - Activities:
+    - At least 20 answers given on coala Gitter rooms
+- The coala all-rounder:
+  - Details: Has met the prerequisites to apply for the coala developer role
+  - Activities:
+    - Introduce `.coafile` in user own repository
+    - Merge a difficulty/newcomer Pull Request
+    - Review at least a difficulty/newcomer Pull Request
+    - Merge a difficulty/low Pull Request
+    - Review at least a difficulty/low or higher Pull Request
+
+So, to be eligible to be a developer at coala, a newcomer must
+complete all the activities mentioned in the `coala all-rounder`
+badge. But in case if some of the newcomers are not able to be
+eligible to apply for coala developer role, they could take away
+the other badges.
+
+Names and metrics of tracked activities will also be shown
+on the leaderboard along with the number of times they
+have been done. That will help maintainers to varify if an
+automatically awarded badge is correct or not.
```

Comment:
I can also think of "levelled" badges, like a badge can have up to X levels. For example a badge could be "Number of PRs reviewed". If you have 5, you get the Level 1 badge. If you have reviewed 20, you get the Level 2 badge and so on and so on...

Reply:
All the things related to reviewing is being done in [meta-review system](https://github.com/li-boxuan/cEPs/blob/e9f7587856e1c236d37e96f54a0cb4d72069776d/cEP-0019.md)

But yeah, we will have more badges while the implementation of this system, these are just a few ideas I have taken to give a better understanding of the design.

Reply:
Agree that 'levels' within a badge is a very common theme in badge design.

Please add a reviewer badge, and a super reviewer badge, to show how the design can be used to provide that type of functionality.